omod/src/main/java/org/openmrs/module/coreapps/page/controller/summarydashboard/ProgramEnrollmentsPageController.java

```diff
@@ -0,0 +1,97 @@
+                             @SpringBean("appFrameworkService") AppFrameworkService appFrameworkService,
+                             PageModel model,
+                             UiSessionContext sessionContext) {
+
+        if (!Context.hasPrivilege(CoreAppsConstants.PRIVILEGE_SUMMARY_DASHBOARD)) {
+            return new Redirect("coreapps", "noAccess", "");
+        }
+
+        model.addAttribute("app", app);
+
+        AppContextModel contextModel = sessionContext.generateAppContextModel();
+        model.addAttribute("appContextModel", contextModel);
+
+        List<Extension> firstColumnFragments = appFrameworkService.getExtensionsForCurrentUser(app.getId() + ".firstColumnFragments", contextModel);
+        Collections.sort(firstColumnFragments);
+        model.addAttribute("firstColumnFragments", firstColumnFragments);
+
+        String programUuid = null;
+        ObjectNode config = app.getConfig();
+        if (config != null) {
+            for (Iterator<Map.Entry<String, JsonNode>> fields = config.getFields(); fields.hasNext();) {
+                Map.Entry<String, JsonNode> field = fields.next();
+                if (StringUtils.equalsIgnoreCase(field.getKey(), "program") ){
+                    programUuid = field.getValue().getTextValue();
+                }
+            }
+        }
+        List<SimpleObject> simpleObjects = new ArrayList<>();
+        if (programUuid != null ) {
+            Program program = Context.getProgramWorkflowService().getProgramByUuid(programUuid);
+            Set<ProgramWorkflow> workflows = program.getWorkflows();
+            if (program != null ) {
+                Date today = Calendar.getInstance().getTime();
+                // retrieve all program enrollments still active as now
+                List<PatientProgram> patientPrograms = Context.getProgramWorkflowService().getPatientPrograms(null, program, null, today, today, null, false);
+                for (PatientProgram patientProgram : patientPrograms) {
+                    Patient patient = patientProgram.getPatient();
+                    for(ProgramWorkflow workflow : workflows) {
+                        PatientState currentState = patientProgram.getCurrentState(workflow);
+                        simpleObjects.add(SimpleObject.create(
+                                "patientUuid", patient.getUuid(),
+                                "patientId", patient.getPatientId(),
+                                "personName", patient.getPersonName().getFamilyName() + ", " + patient.getPersonName().getGivenName(),
+                                "emrId", patient.getPatientIdentifier(emrApiProperties.getPrimaryIdentifierType()),
+                                "dateEnrolled", patientProgram.getDateEnrolled(),
+                                "dateCompleted", patientProgram.getDateCompleted(),
+                                "treatmentState", (currentState != null) ? currentState.getState().getConcept().getName(Context.getLocale()) : ""
```

Comment:
Is there a screen shot for this, and did we test it out on a program with multiple workflows?  Maybe I'm mis-reading, but won't this end up with a row-per-patient-program-workflow, not a row per patient-program? I was thinking that multiple workflows would result in multiple columns, not rows, happy to discuss at standup.

Reply:
@mogoodrich , here is the screenshot, which reflects the mockup (https://pihemr.atlassian.net/browse/SL-671 )

![Screenshot 2024-09-20 at 10 32 35 AM](https://github.com/user-attachments/assets/ba62baf6-5f9a-4ea7-bc02-0a4f2d2cc1cc)

I don't understand how multiple workflows would be displayed in multiple columns? Do you have an example of a program with multiple workflows? Just curious how would that look in the legacy Admin UI, or any other UI if we already have something like this anywhere?

Reply:
Looks like the HIV and Oncology programs both have 2 states, though it looks like they aren't in KGH.  Here's HIV:

![image](https://github.com/user-attachments/assets/6ff689e7-40b3-4a77-b15b-804997db2a10)

So in this case, there are two workflows, "HIV Treatment Status" and "PMTCT Status".  So, ideally, there would be a column for each in the display.

I *think* the current code would create two rows for this program, one with a treatment state of "On ART" and one of "Patient Pregnant", which is definitely not what we want.

We could get away with just combined the two states into a single column, but we've want to name it something generic like "State" instead of "Treatment State".

Reply:
Thanks @mogoodrich ! I got it working with each workflow current state in a separate column. And the translation of the column name was easy by just adding this to the messages.properties:
+ui.i18n.ProgramWorkflow.name.9a3f8252-1588-4f7b-b02c-9e99c437d4ef=Treatment State